py/desitarget/mtl.py

```diff
@@ -412,6 +412,93 @@ def get_mtl_ledger_format():
     return ff
 
 
+def check_archiving(zcatdir=None, mtldir=None):
+    """Check the previous archiving state of tiles.
+
+    Parameters
+    ----------
+    zcatdir : :class:`str`, optional
+        Full path to the directory that hosts redshift catalogs. Defaults
+        to the directory stored in the $ZCAT_DIR environment variable.
+    mtldir : :class:`str`, optional, defaults to ``None``
+        Full path to the directory that hosts the MTL ledgers and the MTL
+        tile file. If ``None``, then look up the MTL directory from the
+        $MTL_DIR environment variable.
+
+    Notes
+    -----
+    - The basic check is that the latest archive date for tiles in the
+      "archive" sub-directory of the redshift catalog directory matches
+      the `ARCHIVEDATE` recorded in the `tiles-specstatus.ecsv` file.
+    - The check is limited to BRIGHT/DARK main survey tiles.
+    - Based on work by Anand Raichoor.
+    """
+    start = time()
+    log.info("Checking archived tiles match tiles-specstatus file...")
+
+    # ADM grab tile files, use default environment variables, if needed.
+    ztilefn = get_ztile_file_name(survey="main")
+    mtldir = get_mtl_dir(mtldir)
+    opsdir = os.path.join(os.path.dirname(mtldir), "ops")
+    ztilefn = os.path.join(opsdir, ztilefn)
+
+    # ADM need to check which tiles are in Main Survey and bright/dark.
+    maintilefn = ztilefn.replace("specstatus", "main")
+    maintiles = Table.read(maintilefn)
+    dorb = (maintiles["PROGRAM"] == "DARK") | (maintiles["PROGRAM"] == "BRIGHT")
+    maintiles = maintiles[dorb]
+
```

Comment:
same, here you d have to adjust for the `obscon` argument, if choosing that approach.

Reply:
Addressed, see my response to your first comment, above.